### ISSUE TITLE
Broken warning & error messages in 'Co-simulation using Cvode' FMU from Dymola 2022x

### ISSUE THREAD

**Issue by wojdatto** [CATIA-Systems/FMPy#384]
Hi,

I noticed that FMU exported with Dymola 2022x with option 'Co-simulation using Cvode' cannot properly display warning and error messages when trying to run it with FMPy.

Rather than the proper warning/error message, '%s' is displayed for each supposed error line:

    [ERROR] %s

My setup:

- FMPy v0.3.7
- Dymola 2022x
- VS 2019 compiler
- Windows 10 Enterprise 64-bit

## How to reproduce

A very simple Modelica model that raises any error needs to be used, e.g.:

```modelica
model ExampleModel
  parameter Real param = 1;
  Real var;
equation
  var = param / 0;
end ExampleModel;
```

I use the following Python code to simulate the FMU:

```python
import shutil

import fmpy
from fmpy.fmi2 import FMU2Slave


def main() -> None:
    fmu_file = "ExampleModel.fmu"
    print(f"Simulate {fmu_file}")
    fmpy.dump(fmu_file)

    model_description = fmpy.read_model_description(fmu_file)

    unzip_dir = fmpy.extract(fmu_file)

    fmu = FMU2Slave(
        guid=model_description.guid,
        unzipDirectory=unzip_dir,
        modelIdentifier=model_description.coSimulation.modelIdentifier,
    )

    sim_time = 0
    step_size = 1
    stop_time = 10

    fmu.instantiate()
    fmu.setupExperiment(startTime=sim_time)
    fmu.enterInitializationMode()
    fmu.exitInitializationMode()

    while sim_time < stop_time:
        fmu.doStep(currentCommunicationPoint=sim_time, communicationStepSize=step_size)
        sim_time += step_size
        print(f"{sim_time = }s")

    fmu.terminate()
    fmu.freeInstance()
    shutil.rmtree(unzip_dir, ignore_errors=False)

if __name__ == "__main__":
    main()
```

Output I receive:

```none
$ python simulate_example_model.py
Simulate ExampleModel.fmu

Model Info

  FMI Version        2.0
  FMI Type           Co-Simulation
  Model Name         ExampleModel
  Description        None
  Platforms          win64
  Continuous States  0
  Event Indicators   0
  Variables          4
  Generation Tool    Dymola Version 2022x (64-bit), 2021-10-08
  Generation Date    2022-03-16T14:48:28Z

Default Experiment

  Stop Time          1.0
  Tolerance          0.0001

Variables (input, output)

  Name               Causality              Start Value  Unit     Description
  CPUtime            output                              s
  EventCounter       output
[ERROR] %s
[ERROR] %s
[ERROR] %s
[ERROR] %s
Traceback (most recent call last):
  File "<snip>\simulate_example_model.py", line 41, in <module>
    main()
  File "<snip>\simulate_example_model.py", line 29, in main
    fmu.exitInitializationMode()
  File "<snip>\venv\lib\site-packages\fmpy\fmi2.py", line 286, in exitInitializationMode
    return self.fmi2ExitInitializationMode(self.component)
  File "<snip>\venv\lib\site-packages\fmpy\fmi2.py", line 215, in w
    raise FMICallException(function=fname, status=res)
fmpy.fmi1.FMICallException: fmi2ExitInitializationMode failed with status 3 (error).
```

On the other hand, when I tried to simulate this FMU using FMPy GUI, it works fine:

![image](https://user-images.githubusercontent.com/49607727/158629725-5769cf52-0157-4afa-b724-50a6d62ec02a.png)

Because of that, I suppose that this is a problem with FMPy rather than with Dymola 2022x FMU export.

It also works fine when I try to create FMU with different configuration, e.g.:

- exporting FMU with the same 'Co-simulation using Cvode' option but in older Dymola version (e.g. 2020)
- using different export options in Dymola 2022x (e.g. 'Co-simulation using Dymola solvers' with Cvode or Radau)

---

[Link to the FMU](https://www.dropbox.com/s/0n0i0rr0st1soso/ExampleModel.fmu?dl=0) with this issue

**Comment by t-sommer** [comment]
If you create the FMU instance, you have to add the logging proxy to use variadic arguments in log messages:

```Python
from ctypes import byref

import shutil

import fmpy
from fmpy.fmi2 import *
from fmpy.logging import addLoggerProxy


def main() -> None:
    fmu_file = r"C:\Users\tsr2\Downloads\ExampleModel.fmu"
    print(f"Simulate {fmu_file}")
    fmpy.dump(fmu_file)

    model_description = fmpy.read_model_description(fmu_file)

    unzip_dir = fmpy.extract(fmu_file)

    callbacks = fmi2CallbackFunctions()
    callbacks.logger         = fmi2CallbackLoggerTYPE(printLogMessage)
    callbacks.allocateMemory = fmi2CallbackAllocateMemoryTYPE(allocateMemory)
    callbacks.freeMemory     = fmi2CallbackFreeMemoryTYPE(freeMemory)

    try:
        addLoggerProxy(byref(callbacks))
    except Exception as e:
        print("Failed to add logger proxy function. %s" % e)

    fmu = FMU2Slave(
        guid=model_description.guid,
        unzipDirectory=unzip_dir,
        modelIdentifier=model_description.coSimulation.modelIdentifier
    )

    sim_time = 0
    step_size = 1
    stop_time = 10

    fmu.instantiate(callbacks=callbacks)
    fmu.setupExperiment(startTime=sim_time)
    fmu.enterInitializationMode()
    fmu.exitInitializationMode()

    while sim_time < stop_time:
        fmu.doStep(currentCommunicationPoint=sim_time, communicationStepSize=step_size)
        sim_time += step_size
        print(f"{sim_time = }s")

    fmu.terminate()
    fmu.freeInstance()
    shutil.rmtree(unzip_dir, ignore_errors=False)


if __name__ == "__main__":
    main()
```

**Comment by wojdatto** [comment]
Thanks, the fixed version seems to work fine!

But should this be necessary? I see that you reopened after a few minutes.

[Here is an FMU](https://www.dropbox.com/s/yet6v6xcesjvymq/ExampleModel_Dymola2020.fmu?dl=0) from the same model compiled with Dymola 2020 with the 'Co-simulation using Cvode' setting:

![image](https://user-images.githubusercontent.com/49607727/159027005-2e51f7ae-39e0-4e14-ac54-0e8f97a9a19a.png)

And it works fine even with my original simple code.

**Comment by t-sommer** [comment]
> But should this be necessary?

No. The issue is resolved with 8273ae8e0acbf6c565712e4145102fa13105c95c.
